### PR TITLE
Updated app to work with new auth server

### PR DESCRIPTION
**Pull request by ccaseypusher**
Updated dependencies, updated auth server and migrated to multipart messages

### CODE REVIEW

File: app/src/main/kotlin/com/pusher/chatkitdemo/ChatKitDemoApp.kt [hunk 1]
```diff
@@ -12,12 +12,13 @@ import com.pusher.platform.logger.LogLevel
...
-import kotlinx.coroutines.experimental.channels.BroadcastChannel
-import kotlinx.coroutines.experimental.channels.Channel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.channels.Channel
 import kotlin.properties.Delegates
 
 private const val INSTANCE_LOCATOR = "v1:us1:05f46048-3763-4482-9cfe-51ff327c3f29"
-private const val TOKEN_PROVIDER_ENDPOINT = "https://chatkit-demo-server.herokuapp.com/token"
+private const val TOKEN_PROVIDER_ENDPOINT = "https://chatkit-demoauth-server.herokuapp.com/token"
```
Comment (daniellevass) [inline app/src/main/kotlin/com/pusher/chatkitdemo/ChatKitDemoApp.kt]: thank you so much for updating this one 🙌

File: app/src/main/kotlin/com/pusher/chatkitdemo/EntryActivity.kt [hunk 2]
```diff
@@ -20,30 +20,33 @@ import kotlinx.android.synthetic.main.activity_entry.*
...
+    @ExperimentalCoroutinesApi
     private val viewModel by lazy { viewModel<EntryViewModel>() }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_entry)
     }
 
+    @ExperimentalCoroutinesApi
+    @ObsoleteCoroutinesApi
```
Comment (daniellevass) [inline app/src/main/kotlin/com/pusher/chatkitdemo/EntryActivity.kt]: do we need both experimental and obsolete annotations on this one?

File: app/src/main/kotlin/com/pusher/chatkitdemo/EntryActivity.kt [hunk 3]
```diff
@@ -64,15 +68,16 @@ class EntryActivity : AppCompatActivity() {
         data class Failure(val error: Error) : State()
     }
 
+    @ExperimentalCoroutinesApi
     private fun State.render() = when (this) {
-        is Idle -> launch(UI) { renderIdle() }
-        is UserIdReady -> launch(UI) {
+        is Idle -> GlobalScope.launch(Dispatchers.Main) { renderIdle() }
```
Comment (daniellevass) [inline app/src/main/kotlin/com/pusher/chatkitdemo/EntryActivity.kt]: Not looked into this properly yet, but will there be any issues with making this GlobalScope?

File: app/src/main/kotlin/com/pusher/chatkitdemo/room/RoomFragment.kt [hunk 4]
```diff
@@ -104,33 +106,58 @@ class RoomFragment : Fragment() {
...
+                                            is Payload.Url -> {
+                                            }
+                                            is Payload.Attachment -> {
+                                            }
+                                            else -> {
+                                            }
+                                        }
+                                    }
+
+                                    Log.d("messageadded", message.sender?.name)
```
Comment (daniellevass) [inline app/src/main/kotlin/com/pusher/chatkitdemo/room/RoomFragment.kt]: should we keep this log message? Should we be adding any more logging to help people get started with this app?

File: build.gradle [hunk 5]
```diff
@@ -2,28 +2,28 @@
...
+        android_support_version = '28.0.0'
         android_arch_version = '1.1.1'
         mockito_version = '2.10.0'
-        constraints_layout_version = '1.1.0'
+        constraints_layout_version = '1.1.3'
         espresso_version = '3.0.1'
         espresso_runner_version = '1.0.1'
         junit_legacy_version = '4.12'
-        chatkit_android_version = '1.1.+'
+        chatkit_android_version = '1.3.3'
```
Comment (daniellevass) [inline build.gradle]: 🙌this is so much better!

File: app/google-services.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (daniellevass) [inline app/google-services.json]: let's remove this file to make it easier for other people to add their own when trying to get this project up and running :)

**Review by daniellevass — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
 
     defaultConfig {
         applicationId "com.pusher.chatkitdemo"
         minSdkVersion 19
-        targetSdkVersion 27
+        targetSdkVersion 28
         versionCode 1
         versionName "1.0"
-
+        multiDexEnabled true
```
[the remark this answers is not in the document]
Comment (ccaseypusher) [reply app/build.gradle]: I couldn't get the app to run without enabling it, but I can retest

File: app/src/main/kotlin/com/pusher/chatkitdemo/EntryActivity.kt (hunk 3 above)
Comment (ccaseypusher) [reply app/src/main/kotlin/com/pusher/chatkitdemo/EntryActivity.kt]: I followed guidance to a do a direct migration from experimental to stable coroutines. So I believe the behaviour is as it was. However there may be a more optimum solution by being more considerate of the scope.

File: app/src/main/kotlin/com/pusher/chatkitdemo/room/RoomFragment.kt (hunk 4 above)
Comment (ccaseypusher) [reply app/src/main/kotlin/com/pusher/chatkitdemo/room/RoomFragment.kt]: Agree that the log message should be removed

**Review by daniellevass — APPROVED** [review]
Looks good to me!

**Pull request merged**